drud/ddev
# Allow configuration to fail when a hook execution fails, fixes #2536

## The Problem/Issue/Bug:
https://github.com/drud/ddev/issues/2536

## How this PR Solves The Problem:
Introduction of a new option (global & local): fail-on-hook-fail

## Manual Testing Instructions:
Easiest way to test is probably by using:
`ddev config global --fail-on-hook-fail=true`
After that, just include a pre-start hook with an exit code (!=0) for instance use the following script saved as test.sh in the config.yaml of your project like this for example:
```
  pre-start:
  - exec-host: bash ./test.sh
```
The contents of the example `test.sh` can be found here (place in root folder of your project)
```
#!/bin/bash
exit 1
```
After you have done all that make sure, you are using the self-build version of `ddev` either from my branch or from the PR, run `ddev poweroff` to ensure, that there are no side-effects from other versions and start your project as usual by calling `ddev start`

## Automated Testing Overview:
`make staticrequired` finished without errors and without warnings.
Manual testing was successful.
Tested:
- inclusion of new option for all projects via `ddev config global --fail-on-hook-fail=true`
- inclusion of new option for specific project via adding `fail_on_hook_fail: true` to projects `.ddev/config.yaml`

## Release/Deployment notes:
Nothing else should be affected, even though this was my first project using "Go" 🙂
